### PR TITLE
removing most v2 references from source/

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: Low
Testing: n/a
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: source/extensions/tracers/zipkin/zipkin_tracer_impl.h [hunk 1]
```diff
@@ -163,7 +163,7 @@ struct CollectorInfo {
   std::string endpoint_{DEFAULT_COLLECTOR_ENDPOINT};
 
   // The version of the collector. This is related to endpoint's supported payload specification and
-  // transport. Currently it defaults to envoy::config::trace::v2::ZipkinConfig::HTTP_JSON_V1. In
+  // transport. Currently it defaults to envoy::config::trace::32::ZipkinConfig::HTTP_JSON_V1. In
```
Comment (daixiang0) [inline source/extensions/tracers/zipkin/zipkin_tracer_impl.h:166]: typo here

File: test/common/config/delta_subscription_test_harness.h [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 #include <queue>
 
+#include "envoy/api/v2/discovery.pb.h"
```
Comment (daixiang0) [inline test/common/config/delta_subscription_test_harness.h:5]: still use v2?

Comment (alyssawilk) [reply test/common/config/delta_subscription_test_harness.h:5]: yeah this PR only fixes src/  :-)

File: source/extensions/tracers/zipkin/zipkin_tracer_impl.h (hunk 1 above)
Comment (alyssawilk) [reply source/extensions/tracers/zipkin/zipkin_tracer_impl.h]: nice catch, fixed!

**Review by snowp — APPROVED** [review]
Thanks!

**Comment by alyssawilk** [comment]
unfortunately windows CI failure is real, and due to https://github.com/envoyproxy/envoy/blob/main/source/server/proto_descriptors.cc#L13
@mattklein123 do you know if I can just remove the validate function?  I'm not clear on the v3 equivalents here.

**Comment by mattklein123** [comment]
> @mattklein123 do you know if I can just remove the validate function? I'm not clear on the v3 equivalents here.

Hmm. I don't remember the history here. I think this is to make linking issues obvious? We should probably swap this to v3 equivalents and leave it? I guess it would be OK to just remove it all for now and see if it bites us again. I'm fine either way.

**Comment by alyssawilk** [comment]
yeah removing them breaks things and I'm not sure if they're real or test breakages. I'll err on the side of caution and just update to v3 equivalents.
/wait

**Review by htuch — DISMISSED** [review]
LGTM, thanks!

**Comment by alyssawilk** [comment]
ugh this windows build is not having it :-(
/wait

**Comment by alyssawilk** [comment]
@wrowe do you have any ideas why this PR is failing on windows only?  @davinci26 had suggested linking the variou removed files into the failing tests and I believe I've done that to no avail :-/

**Comment by davinci26** [comment]
@alyssawilk I think that the issue is the test that are currently failing are including `protobuf_link_hacks.h` so you will need to add/replace with `v2_link_hack.h` include to those tests files as well. I am not sure if just adding a reference to the package without including is good enough. I have verified it locally as well

I think that the reason this is happening is because the compiler\linker thinks that this dependency is not used and it does not link it with the test. However, I do not fully understand why this is only happening on Windows. I have seen it a couple of times so I can recognize it as a pattern. @wrowe might be able to give more details.

**Comment by alyssawilk** [comment]
Man, that took longer than expected.  Finally got windows happy.  Thanks @davinci26 for the pointer to force-includes, that seems to have done the trick!

**Review by snowp — APPROVED** [review]
Thanks!

**Review by davinci26 — APPROVED** [review]
happy to help, sorry if this was frustrating

**Pull request merged**